src/OpenCV/ImgProc/GeometricImgTransform.hsc

```diff
@@ -297,3 +298,37 @@ getRotationMatrix2D center angle scale = unsafeCoerceMat $ unsafePerformIO $
   where
     c'angle = realToFrac angle
     c'scale = realToFrac scale
+
+{- | Applies a generic geometrical transformation to an image.
+
+<http://docs.opencv.org/3.0-last-rst/modules/imgproc/doc/geometric_transformations.html#remap>
+-}
+remap
+    :: Mat ('S [inputHeight, inputWidth]) inputChannels inputDepth
+    -> Mat ('S [outputHeight, outputWidth]) ('S 2) ('S Float)
+    -> InterpolationMethod
+    -> BorderMode
+    -> Mat ('S [outputHeight, outputWidth]) inputChannels inputDepth
```

Comment:
How sure are you that `cv::remap` will not throw an exception?

The [C++ implementation](https://github.com/opencv/opencv/blob/master/modules/imgproc/src/imgwarp.cpp#L4850) has a couple of assertions. I haven't checked any of the functions called by `cv::remap` itself, but they might also throw exceptions.

We must wrap the result of `remap` in `CvException`.